src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -533,6 +533,29 @@ protected override void GetDependenciesDueToMethodCodePresenceInternal(ref Depen
             }
         }
 
+        public override bool HasConditionalDependenciesDueToMethodGenericDictionary(MethodDesc method)
+        {
+            return !IsReflectionBlocked(method);
+        }
+
+        public override void GetConditionalDependenciesDueToMethodGenericDictionary(ref CombinedDependencyList dependencies, NodeFactory factory, MethodDesc method)
+        {
+            if ((_generationOptions & UsageBasedMetadataGenerationOptions.CreateReflectableArtifacts) == 0
```

Comment:
Nit: I would probably add an assert that the `method` is specific instantiation (not an open generic), otherwise would add a conditional dependency between the same node (basically if `typicalMethod == method`).
I know it should never happen, but it would be nice as as sort of documentation.